pkg/job/job.go

```diff
@@ -288,20 +289,35 @@ func (k *kubeJob) uploadImageJob(jobName, imageName string) *batch.Job {
 func (k *kubeJob) UploadBaseISO() error {
 	ctx := context.Background()
 	log := logutil.FromContext(ctx, k.log)
+	jobName, err := uploadBaseISOJobName()
```

Comment:
You may just want to use this instead: https://github.com/kubernetes/apiserver/blob/v0.19.2/pkg/storage/names/generate.go#L49

Reply:
Or perhaps you can just set the Job's `GenerateName` field instead of the `Name` field, and let the api server do this work.